exec/java-exec/src/main/codegen/templates/Decimal/DecimalFunctions.java

```diff
@@ -170,6 +170,8 @@ public void eval() {
               left.precision, left.scale,
               right.precision, right.scale);
 
+      // Do the verification of computed scale and precision here based on actual data
+      typeInference.verifyScaleAndPrecision();
```

Comment:
I don't see moving the verification check here rather than doing in planning time helps. Since for each addition call we still increase the precision by 1, hence for 39 such additions precision will go beyond max allowed 38 and scale will become -1, hence throwing the error.